testsuite/MDAnalysisTests/parallelism/test_multiprocessing.py

```diff
@@ -0,0 +1,209 @@
+# MDAnalysis: A Python package for the rapid analysis of molecular dynamics
+# simulations. In S. Benthall and S. Rostrup editors, Proceedings of the 15th
+# Python in Science Conference, pages 102-109, Austin, TX, 2016. SciPy.
+# doi: 10.25080/majora-629e541a-00e
+#
+# N. Michaud-Agrawal, E. J. Denning, T. B. Woolf, and O. Beckstein.
+# MDAnalysis: A Toolkit for the Analysis of Molecular Dynamics Simulations.
+# J. Comput. Chem. 32 (2011), 2319--2327, doi:10.1002/jcc.21787
+#
+import multiprocessing
+
+import numpy as np
+import pytest
+import pickle
+from numpy.testing import assert_equal
+
+import MDAnalysis as mda
+from MDAnalysis.lib.picklable_file_io import pickle_open
+from MDAnalysis.coordinates.core import get_reader_for
+
+from MDAnalysisTests.datafiles import (
+    CRD,
+    PSF, DCD,
+    DMS,
+    DLP_CONFIG,
+    DLP_HISTORY,
+    INPCRD,
+    GMS_ASYMOPT,
+    GMS_SYMOPT,
+    GRO,
+    GSD,
+    GSD_long,
+    LAMMPSdata_mini,
+    LAMMPSDUMP,
+    mol2_molecules,
+    MMTF,
+    NCDF,
+    PDB, PDB_small, PDB_multiframe,
+    PDBQT_input,
+    PQR,
+    TRR,
+    TRJ,
+    TRZ,
+    TXYZ,
+    XTC,
+    XPDB_small,
+    XYZ_mini, XYZ, XYZ_bz2,
+)
+
+
+def textio_line(file, i):
+    return file.readlines()[i]
+
+
+def test_multiprocess_fileio():
+    p = multiprocessing.Pool(2)
+    with pickle_open(PDB) as PDB_file:
+        ref = PDB_file.readlines()[:4]
+    with pickle_open(PDB) as PDB_file:
+        res = np.array([p.apply(textio_line, args=(PDB_file, i))
+                        for i in range(4)])
+    p.close()
+    assert_equal(res, ref)
+
+
+@pytest.fixture(params=[
+    (PSF, DCD),
+    (GRO, XTC),
+    (PDB_multiframe,),
+    (XYZ,),
+    (XYZ_bz2,),  # .bz2
+    (GMS_SYMOPT,),  # .gms
+    (GMS_ASYMOPT,),  # .gz
+    (GSD_long,),
+    (NCDF,),
+    (np.arange(150).reshape(5, 10, 3).astype(np.float64),),
+    (GRO, [GRO, GRO, GRO, GRO, GRO]),
+    (PDB, [PDB, PDB, PDB, PDB, PDB]),
+    (GRO, [XTC, XTC]),
+])
+def u(request):
+    if len(request.param) == 1:
+        f = request.param[0]
+        return mda.Universe(f)
+    else:
+        top, trj = request.param
+        return mda.Universe(top, trj)
+
+
+# Define target functions here
+# inside test functions doesn't work
+def cog(u, ag, frame_id):
+    u.trajectory[frame_id]
+
+    return ag.center_of_geometry()
+
+
+def getnames(u, ix):
+    # Check topology stuff works
+    return u.atoms[ix].name
+
+
+def test_trajectory_next(u):
+    u.trajectory.next()
+    u_p = pickle.loads(pickle.dumps(u))
+    u.trajectory.next()
+    u_p.trajectory.next()
+    assert_equal(u.atoms.positions[0], u_p.atoms.positions[0])
+
+
+def test_multiprocess_COG(u):
+    ag = u.atoms[2:5]
+
+    ref = np.array([cog(u, ag, i)
+                    for i in range(3)])
+
+    p = multiprocessing.Pool(2)
+    res = np.array([p.apply(cog, args=(u, ag, i))
+                    for i in range(3)])
+    p.close()
+    assert_equal(ref, res)
+
+
+def test_multiprocess_names(u):
```

Comment:
What do these tests tell us? We already test that the reader and universe can be pickled. Multiprocessing does not add much to this. Plus the tests take a long time looking at the build log.

Reply:
I think this is specific to test `AtomGroup` can be pickled with `Universe` successfully. The difference between direct pickle and multiprocessing is that, in multiprocessing, it's happening all in another process so it does not have the old `Universe` to be anchored on. 

I think it makes sense to only test it on one case, it shouldn't differ from one reader to another.

Reply:
yeah once is enough to test this

Reply:
making sure it happens in another process is also a good idea given the anchors. However, the test name should reflect that, i.e. `test_universe_unpickle_in_new_process`.  Using multiprocessing is a convenient way to do this.